editor/components/post-featured-image/index.js

```diff
@@ -84,6 +90,7 @@ const applyWithAPIData = withAPIData( ( { featuredImageId } ) => {
 
 	return {
 		media: `/wp/v2/media/${ featuredImageId }`,
+		postType: '/wp/v2/types/post?context=edit',
```

Comment:
We shouldn't assume the "post" type, since the editor can be used for custom post types.

`withAPIData` passes as a second argument an object containing a helper you can use here: `type`

See: https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-api-data

Reply:
> We shouldn't assume the "post" type, since the editor can be used for custom post types.

Nice point, an improvement was made so we take into consideration the post type.

> .withAPIData passes as a second argument an object containing a helper you can use here: type
See: https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-api-data

This helper receives that the type and helps us create URLs for requests related to that type e.g: post -> posts, page -> pages. In this case, we need to use directly the post type name so I’m using the result of selector getCurrentPostType.